code/ATMOSPHERICS/atmospherics.dm

```diff
@@ -53,6 +55,7 @@ GLOBAL_DATUM_INIT(pipe_icon_manager, /datum/pipe_icon_manager, new())
 		stored = new(src, make_from = src)
 
 /obj/machinery/atmospherics/Destroy()
+	update_all_pipe_vision()
```

Comment:
This one seems to be something that should be done at the end of the Destroy. Just before calling `..()`